engine/src/main/java/io/zeebe/engine/processor/StreamProcessor.java

```diff
@@ -272,11 +272,23 @@ public boolean isFailed() {
   }
 
   public ActorFuture<Long> getLastProcessedPositionAsync() {
-    return actor.call(processingStateMachine::getLastSuccessfulProcessedEventPosition);
+    return actor.call(
+        () -> {
+          if (processingStateMachine != null) {
+            return processingStateMachine.getLastSuccessfulProcessedEventPosition();
+          }
+          return null;
+        });
   }
 
   public ActorFuture<Long> getLastWrittenPositionAsync() {
-    return actor.call(processingStateMachine::getLastWrittenEventPosition);
+    return actor.call(
+        () -> {
+          if (processingStateMachine != null) {
+            return processingStateMachine.getLastWrittenEventPosition();
+          }
+          return null;
```

Comment:
I could image that it might make more sense to throw here a `IllegalStateException`. I would assume that the future is then complete exceptionally and on the other side (AsyncSnapshotDirector) you already handle the exceptional case so no changes needed there

Reply:
Then what's the practical difference to the previous code? We would still log an exception that shows up during fast tests

Reply:
Yes but it was not clear what the error is, then it is :sweat_smile: